lib/core/utils/preload.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * Validated the preload object
+ * @param {Object | boolean} preload configuration object or boolean passed via the options parameter to axe.run
+ * @return {boolean}
+ * @private
+ */
+function isValidPreloadObject(preload) {
+	return (
+		typeof preload === 'object' &&
+		preload.hasOwnProperty('assets') &&
+		Array.isArray(preload.assets)
+	);
+}
+
+/**
+ * Returns a boolean which decides if preload is configured
+ * @param {Object} options run configuration options (or defaults) passed via axe.run
+ * @return {boolean}
+ */
+axe.utils.shouldPreload = function shouldPreload(options) {
+	if (!options || !options.preload) {
+		return false;
+	}
+	if (typeof options.preload === 'boolean') {
+		return options.preload;
+	}
+	return isValidPreloadObject(options.preload);
+};
+
+/**
+ * Constructs a configuration object representing the preload requested assets & timeout
+ * @param {Object} options run configuration options (or defaults) passed via axe.run
+ * @return {Object}
+ */
+axe.utils.getPreloadConfig = function getPreloadConfig(options) {
+	// default fallback configuration
+	const config = {
+		assets: axe.constants.preloadAssets,
+		timeout: axe.constants.preloadAssetsTimeout
+	};
+
+	// if type is boolean
+	if (typeof options.preload === 'boolean') {
+		return config;
+	}
+
+	// if type is object - ensure an array of assets to load is specified
+	if (!isValidPreloadObject(options.preload)) {
+		throw new Error(
+			'No assets configured for preload in aXe run configuration'
+		);
+	}
+
+	// check if requested assets to preload are valid items
+	const areRequestedAssetsValid = options.preload.assets.every(a =>
+		axe.constants.preloadAssets.includes(a.toLowerCase())
+	);
+
+	if (!areRequestedAssetsValid) {
+		throw new Error(
+			`Requested assets, not supported by aXe.` +
```

Comment:
IIRC we're not capitalizing the "x" anymore.

Reply:
to avoid `i18n` concerns removed aXe.